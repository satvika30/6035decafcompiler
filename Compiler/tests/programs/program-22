class Program {
	int a[10], temp[10], c[10];
	int num_trials;
	
	void shift(int rot_size) {
		int lcv;
		if (rot_size < 0) {
			rot_size = 10-((-rot_size)%10);
		}
		for lcv = 0, 10 {
			temp[(lcv+rot_size)%10] = a[lcv];
		}
		for lcv = 0, 10 {
			c[lcv] = temp[lcv];
		}
	}
	
	// print values
	void print_array() {
		int i;
		for i = 0, 10 {
			callout("printf","%d ",c[i]);
		}
		callout("printf","\n");
	}
	
	void main() {
		int i, rotation_size;
		num_trials = 10;
		// initialize array
		callout("printf","Initial array: \n");
		for i = 0, 10 {
			a[i] = callout("random");
			callout("printf","%d ",a[i]);
		}
		callout("printf","\n");
		for i = -10, num_trials {
			rotation_size = i;
			callout("printf", "rotation of %d:\n", i);
			shift(rotation_size);
			print_array();
		}
	}
}