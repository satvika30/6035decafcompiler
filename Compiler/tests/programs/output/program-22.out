Initial array: 
0 1 2 3 4 5 6 7 8 9 
rotation of -10:
0 1 2 3 4 5 6 7 8 9 
rotation of -9:
9 0 1 2 3 4 5 6 7 8 
rotation of -8:
8 9 0 1 2 3 4 5 6 7 
rotation of -7:
7 8 9 0 1 2 3 4 5 6 
rotation of -6:
6 7 8 9 0 1 2 3 4 5 
rotation of -5:
5 6 7 8 9 0 1 2 3 4 
rotation of -4:
4 5 6 7 8 9 0 1 2 3 
rotation of -3:
3 4 5 6 7 8 9 0 1 2 
rotation of -2:
2 3 4 5 6 7 8 9 0 1 
rotation of -1:
1 2 3 4 5 6 7 8 9 0 
rotation of 0:
0 1 2 3 4 5 6 7 8 9 
rotation of 1:
9 0 1 2 3 4 5 6 7 8 
rotation of 2:
8 9 0 1 2 3 4 5 6 7 
rotation of 3:
7 8 9 0 1 2 3 4 5 6 
rotation of 4:
6 7 8 9 0 1 2 3 4 5 
rotation of 5:
5 6 7 8 9 0 1 2 3 4 
rotation of 6:
4 5 6 7 8 9 0 1 2 3 
rotation of 7:
3 4 5 6 7 8 9 0 1 2 
rotation of 8:
2 3 4 5 6 7 8 9 0 1 
rotation of 9:
1 2 3 4 5 6 7 8 9 0 